src/routes/search.ts

```diff
@@ -0,0 +1,89 @@
+import type { FastifyPluginCallback } from 'fastify';
+import youtubedl from 'youtube-dl-exec';
+
+const searchTrack: FastifyPluginCallback = async (fastify): Promise<void> => {
+	fastify.post<{ Body: { search?: string } }>(
+		'/search',
+		{
+			schema: {
+				description: 'Search for a track',
+				body: {
+					type: 'object',
+					required: ['search'],
+					properties: {
+						search: {
+							description: 'The search query',
+							type: 'string',
+						},
+					},
+				},
+				response: {
+					200: {
+						description: 'Successful response for searching a track',
+						type: 'object',
+						properties: {
+							search: {
+								type: 'array',
+								items: {
+									type: 'object',
+									properties: {
+										artist: {
+											description: 'The artist of the track, if any',
+											type: 'string',
+										},
+										duration: {
+											description: 'The duration of the track',
+											type: 'number',
+										},
+										description: {
+											description: 'The description of the track, if any',
+											type: 'string',
+										},
+										title: {
+											description: 'The titel of the track',
+											type: 'string',
+										},
+										thumbnail: {
+											description: 'The thumbnail of the track',
+											type: 'string',
+										},
+										url: {
+											description: 'The url of the track',
+											type: 'string',
+										},
+									},
+								},
+							},
+						},
+					},
+					400: {
+						description: 'Failure response by not providing a search query',
+						type: 'object',
+					},
+				},
+			},
+		},
+		async (req, reply) => {
+			const search = req.body.search;
+			if (!search) {
+				return reply.code(400).send();
+			}
+
+			// @ts-ignore
+			const { entries } = await youtubedl(`ytsearch10:${search}`, { dumpSingleJson: '' });
```

Comment:
Why is it ignored? What's the `10` in the search? Is it not possible to search in SoundCloud as well like Lavalink does?

Reply:
1) The types are wrong/don't support `dumpSingleJson`.
2) `n` refers to the number of results wanted.
3) Probably could, but I wanted to avoid the user doing `ytsearchN` themselves, so there needs to be a nicer way to handle this then.